cpp/include/raft/sparse/linalg/sddmm.cuh

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/sparse/linalg/detail/sddmm.hpp>
+#include <raft/sparse/linalg/detail/utils.cuh>
+
+namespace raft {
+namespace sparse {
+namespace linalg {
+
+/**
+ * @brief This function performs the multiplication of dense matrix A and dense matrix B,
+ * followed by an element-wise multiplication with the sparsity pattern of C.
+ * It computes the following equation: C = alpha · (A * B ∘ spy(C)) + beta · C
+ * where A,B are device matrix views and C is a CSR device matrix view
+ *
+ * @tparam ValueType Data type of input/output matrices (float/double)
+ * @tparam IndexType Type of C
+ * @tparam LayoutPolicyA layout of A
+ * @tparam LayoutPolicyB layout of B
+ * @tparam NZType Type of Cz
+ *
+ * @param[in] handle raft handle
+ * @param[in] trans_a transpose operation for A
+ * @param[in] trans_b transpose operation for B
+ * @param[in] alpha scalar
+ * @param[in] a input raft::device_matrix_view
+ * @param[in] b input raft::device_matrix_view
+ * @param[in] beta scalar
+ * @param[out] c output raft::device_csr_matrix_view
+ */
+template <typename ValueType,
+          typename IndexType,
+          typename NZType,
+          typename LayoutPolicyA,
+          typename LayoutPolicyB>
+void sddmm(raft::resources const& handle,
```

Comment:
That makes sense! But considering the minimum of the customer's learning cost, may I suggest here to keep the params in a similar sequence with the `cuSparse` original API？

Reply:
The RAFT APIs don't assume the user is familiar with, nor even knows that cusparse is being used under the hood, so it's best to keep up with the conventions established by RAFT so that the user has a consistent experience. 

@lowener is correct to point this out- we had a lot of discussions about this when the convention was established and we strive to use the same conventions everywhere. 

(Resources, param structs, in, out, params)

Reply:
Fixed it.